### PR TITLE
Bugfix/337 338 tribal bogus url issues

### PR DESCRIPTION
**Pull request by cschwinderg**
## Related Issues:
* [HMW-337](https://jira.epa.gov/browse/HMW-337)
* [HMW-338](https://jira.epa.gov/browse/HMW-338)
* [HMW-339](https://jira.epa.gov/browse/HMW-339)

## Main Changes:
* Added an error message to the `state-and-tribal` page if a user includes a nonexistent state or tribe in the URL.
* Investigated an issue with the back button not working on the tribe page after navigating to a bad url.
    * I was unable to reproduce this locally. I'm hoping maybe the above change fixed this. If it doesn't 
* Removed an unnecessary sentece from the state/tribe page for when there are no documents.

## Steps To Test 337:
1. Navigate to http://localhost:3000/tribe/ABCDEF
2. Verify the `Data is not available for this location. Please select a state, territory or tribe from the dropdown below.` message is displayed
3. Select a state or tribe from the dropdown and click Go
4. Verify the message in step 2 is no longer displayed
5. Navigate to http://localhost:3000/state/ZZ/water-quality-overview
6. Repeat steps 2 - 4
7. Navigate to http://localhost:3000/state/AL/nonexistent-tab
8. Verify the `Invalid URL path. Please select a state, territory or tribe from the dropdown below.` message is displayed
9. Select a state from the dropdown and click Go
10. Verify the message in step 8 is no longer displayed

## Steps To Test 338:
1. Navigate to http://localhost:3000/tribe/DELAWARENATION
2. In the same tab navigate to http://localhost:3000/tribe/ABCDEF
3. Click the back button on the browser
4. Verify the url changes to http://localhost:3000/tribe/DELAWARENATION
5. Verify the data for `DELAWARENATION` is displayed

## Steps To Test 339:
1. Navigate to http://localhost:3000/tribe/DELAWARENATION
2. Expand the documents section
3. Verify the `Select a document below to download a copy of the report.` text is not visible
4. Navigate to http://localhost:3000/state/KS/water-quality-overview
5. Verify the `Select a document below to download a copy of the report.` text is visible, only above the `Documents Related to Integrated Report` section

### CODE REVIEW

File: app/client/src/components/pages/StateTribal.js [hunk 1]
```diff
@@ -178,13 +178,24 @@ function StateTribal() {
...
+
   // redirect to '/stateandtribe' if the url is /state or /tribe
   useEffect(() => {
     const pathname = window.location.pathname.toLowerCase();
     if (['/state', '/state/', '/tribe', '/tribe/'].includes(pathname)) {
+      setErrorCode('');
       navigate('/state-and-tribal', { replace: true });
     }
-  }, [navigate]);
+  }, [navigate, setErrorCode]);
```
Comment (maxdiebold-erg) [inline app/client/src/components/pages/StateTribal.js:198]: Typo here, `setErrorCode` should be `setErrorType`.

**Review by maxdiebold-erg — CHANGES_REQUESTED** [review]
Works like a charm! I found one typo that will throw a reference error, though, in one of the edge cases.

File: app/client/src/components/pages/StateTribal.js (hunk 1 above)
Comment (maxdiebold-erg) [reply app/client/src/components/pages/StateTribal.js]: I'll go ahead and make the fix!

**Review by maxdiebold-erg — APPROVED** [review]

File: app/client/src/components/pages/StateTribal.js (hunk 1 above)
Comment (cschwinderg) [reply app/client/src/components/pages/StateTribal.js]: Good catch! Thank you for fixing that as well!

**Pull request merged**